vulture_os/toolkit/api_parser/perception_point_x_ray/perception_point_x_ray.py

```diff
@@ -0,0 +1,237 @@
+
+            try:
+                response = self.session.get(
+                    url,
+                    params=query,
+                    timeout=timeout,
+                    proxies=self.proxies,
+                    verify=self.api_parser_custom_certificate or self.api_parser_verify_ssl
+                )
+            except requests.exceptions.Timeout:
+                logger.debug(f"[{__parser__}]:__execute_query: TimeoutError: retry += 1 (waiting 10s)",
+                             extra={'frontend': str(self.frontend)})
+                retry += 1
+                self.evt_stop.wait(10.0)
+                continue
+            except Exception as e:
+                logger.error(f"[{__parser__}]:__execute_query: Exception: {e}", extra={'frontend': str(self.frontend)})
+                raise PerceptionPointXRayAPIError(f"[{__parser__}]:__execute_query: Error at request. Exception: {e}")
+            else:
+                if response.status_code == 200:
+                    try:
+                        return response.json()
+                    except json.decoder.JSONDecodeError as e:
+                        raise PerceptionPointXRayAPIError(
+                            f"[{__parser__}]:__execute_query: Error at JSON decoding. Exception JSONDecodeError: {e}") from e
+
+                elif response.status_code == 429:
+                    logger.error(
+                        f"[{__parser__}]:__execute_query: Error at PerceptionPointXRay API Call URL: {url} Code: 429."
+                        f"Wait a minute before continue. (retry {retry})",
+                        extra={'frontend': str(self.frontend)})
+                    retry += 1
+                    self.evt_stop.wait(60.0)
+                    continue
+
+                else:
+                    logger.error(f"[{__parser__}]:__execute_query: Error at PerceptionPointXRay API Call URL: {url} Code: {response.status_code} Content: {response.content}\n" \
+                        f"Response : {response}", extra={'frontend': str(self.frontend)})
+                    retry += 1
+                    self.evt_stop.wait(10.0)
+                    continue
+
+        msg = f"Error at PerceptionPointXRay API Call URL: {url}: Unable to get a correct response"
+        logger.error(f"[{__parser__}]:__execute_query: {msg}", extra={'frontend': str(self.frontend)})
+        raise PerceptionPointXRayAPIError(msg)
+
+
+
+    def _connect(self):
+        if self.session is None:
+            self.session = requests.Session()
+            self.session.headers.update(self.HEADERS)
+            self.session.headers.update({"Authorization": f"Token {self.token}"})
+            self.session.headers.update({"User-Agent": ""})
+        return True
+
+    def test(self):
+        try:
+            if self.session is None:
+                self._connect()
+
+            since = timezone.now() - timedelta(days=30)
```

Comment:
Please avoid querying on such large data ranges: the query will be too long/big for the efficient testing on GUI